File: lib/src/main/java/com/tustar/action/Ch9.java
package com.tustar.action;

import io.reactivex.ObservableTransformer;
import io.reactivex.functions.Function;

public class Ch9 {

    public static ObservableTransformer<Integer, String>transformer() {
        return upstream -> upstream.map((Function<Integer, String>) integer -> {
            return String.valueOf(integer);
        });
    }
}

File: lib/src/main/java/com/tustar/action/Ch9.kt
package com.tustar.action

import io.reactivex.Observable


fun main(args: Array<String>) {
    Observable.just(123, 456)
            .compose(transformer())
            .subscribe {
                println("s=$it")
            }
}

private fun transformer(): (Observable<Int>) -> Observable<String> {
    return { upstream ->
        upstream.map { "[$it]" }
    }
}
